sql/core_clients_last_seen_raw_v1.sql

```diff
@@ -0,0 +1,61 @@
+  -- Equivalent to, but more efficient than, calling udf_bitmask_range(1, 28)
+CREATE TEMP FUNCTION
+  udf_bitmask_lowest_28() AS (0x0FFFFFFF);
+  --
+CREATE TEMP FUNCTION
+  udf_shift_one_day(x INT64) AS (IFNULL((x << 1) & udf_bitmask_lowest_28(),
+      0));
+  --
+CREATE TEMP FUNCTION
+  udf_combine_days(prev INT64,
+    curr INT64) AS (udf_shift_one_day(prev) + IFNULL(curr,
+      0));
+  --
+WITH
+  _current AS (
+  SELECT
+    -- In this raw table, we capture the history of activity over the past
+    -- 28 days for each usage criterion as a single 64-bit integer. The
+    -- rightmost bit represents whether the user was active in the current day.
+    CAST(TRUE AS INT64) AS days_seen_bits,
+    -- We only trust profile_date if it is within one week of the ping submission,
+    -- so we ignore any value more than seven days old.
+    IF(DATE_DIFF(submission_date, profile_date, DAY) BETWEEN 0 AND 6,
+      1 << DATE_DIFF(submission_date, profile_date, DAY),
+      0) AS days_created_profile_bits,
+    * EXCEPT (submission_date)
+  FROM
+    core_clients_daily_v1
+  WHERE
+    submission_date = @submission_date ),
+  --
+  _previous AS (
+  SELECT
+    * EXCEPT (submission_date)
+  FROM
+    core_clients_last_seen_raw_v1 AS cls
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    -- Filter out rows from yesterday that have now fallen outside the 28-day window.
+    AND udf_shift_one_day(days_seen_bits) > 0)
+  --
+SELECT
+  @submission_date AS submission_date,
+IF
+  (_current.client_id IS NOT NULL,
+    _current,
+    _previous).* REPLACE ( --
+    udf_combine_days(_previous.days_seen_bits,
+    _current.days_seen_bits) AS days_seen_bits,
```

Comment:
the web console formatted this with one more indent level on this line for me
```suggestion
      _current.days_seen_bits) AS days_seen_bits,
```